core/trino-main/src/main/java/io/trino/execution/buffer/PageSerializer.java

```diff
@@ -94,13 +92,13 @@ private static class SerializedPageOutput
             extends SliceOutput
     {
         private static final int INSTANCE_SIZE = instanceSize(SerializedPageOutput.class);
-        // TODO: implement getRetainedSizeInBytes in Lz4Compressor
-        private static final int COMPRESSOR_RETAINED_SIZE = toIntExact(instanceSize(Lz4Compressor.class) + sizeOfIntArray(Lz4RawCompressor.MAX_TABLE_SIZE));
+        // TODO: implement getRetainedSizeInBytes in Lz4Compressor and ZstdCompressor
```

Comment:
this is prerequisite - otherwise we regress on accuracy of memory management.

Reply:
> this is prerequisite - otherwise we regress on accuracy of memory management.

I have added back the computation of Lz4RawCompressor.MAX_TABLE_SIZE, thanks for the advice~